src/raftstore/store/worker/region.rs

```diff
@@ -104,6 +105,37 @@ impl SnapContext {
         let raft_db = self.raft_db.clone();
         let raw_snap = Snapshot::new(self.kv_db.clone());
 
+        let mut old_snaps = None;
+        while self.mgr.get_total_snap_size() > self.mgr.max_total_snap_size() {
+            if old_snaps.is_none() {
+                let snaps = match self.mgr.list_idle_snap() {
+                    Ok(snaps) => snaps,
+                    Err(_) => break,
+                };
+                let mut key_and_snaps = Vec::with_capacity(snaps.len());
+                for (key, is_sending) in snaps {
+                    if !is_sending {
+                        continue;
+                    }
+                    let snap = match self.mgr.get_snapshot_for_sending(&key) {
+                        Ok(snap) => snap,
+                        Err(_) => continue,
+                    };
+                    if !snap.meta().is_ok() {
+                        continue;
+                    }
+                    key_and_snaps.push((key, snap));
+                }
+                key_and_snaps
+                    .sort_by_key(|item| Reverse(item.1.meta().unwrap().modified().unwrap()));
+                old_snaps = Some(key_and_snaps);
+            }
+            match old_snaps.as_mut().unwrap().pop() {
+                Some((key, snap)) => self.mgr.delete_snapshot(&key, snap.as_ref(), false),
+                None => return Err(Error::TooManySnapshots),
```

Comment:
why TooManySnapshots here?

Reply:
Here means there are too many snapshot files, and none of them can be deleted for now. So we return `TooManySnapshots`.